.ci/scripts/coverity.sh

```diff
@@ -0,0 +1,139 @@
+COV_ANALYSE_OPTIONS+=" --checker-option RESOURCE_LEAK:allow_unimpl:true"
+
+function show_usage() {
+    echo ""
+    echo "Usage $(basename "$0") [options]"
+    echo ""
+    echo "Run static code analysis using Coverity, generate a defects report, and compare"
+    echo "the current report with the known issues filter file."
+    echo ""
+    echo "Script should be run from the main directory after makefiles have been generated."
+    echo ""
+    echo "Options:"
+    echo "--cov-analyze-args <args>  Additional arguments for cov-analyze."
+    echo "                           For more information, see cov-analyze manual."
+    echo ""
+    echo "--skip-coverity            Skip the Coverity stage."
+    echo "--skip-cov-build           Skip cov-build in the Coverity stage."
+    echo "                           filters out at least one issue."
+    echo ""
+    echo "-h, --help                 Show usage."
+    echo ""
+}
+
+# Build and run static code analysis with Coverity
+function build_with_coverity() {
+    echo "Building target with Coverity..."
+
+    if [ ! -f Makefile ]; then
+        echo "ERROR: Make file not found"
+        echo "Need to run from a directory with Make files"
+        return 1
+    fi
+
+    make clean >/dev/null
+    # Run cov-build
+    cov-build --dir "${COV_BUILD_DIR}" make $make_opt all >/dev/null
+    err_code=$?
+    return $err_code
+}
+
+# Run Coverity analysis
+function run_coverity_analysis() {
+    echo "Running Coverity analysis..."
+
+    # Run cov-analyze
+    # shellcheck disable=SC2086
+    cov-analyze --dir $COV_BUILD_DIR "$@" --jobs auto $COV_ANALYSE_OPTIONS
+    err_code=$?
+    return $err_code
+}
+
+# Main function
+function main() {
+    # Parse command-line options
+    while [[ $# -gt 0 ]]; do
+        case "$1" in
+        --cov-analyze-args)
+            shift
+            COV_ANALYZE_ARGS="$1"
+            ;;
+        --skip-coverity)
+            SKIP_COVERITY=true
+            ;;
+        --skip-cov-build)
+            SKIP_COV_BUILD=true
+            ;;
+        -h | --help)
+            show_usage
+            exit 0
+            ;;
+        *)
+            echo "Unrecognized option: $1"
+            show_usage
+            exit 1
+            ;;
+        esac
+        shift
+    done
+
+    # Build with Coverity
+    if [ -z "$SKIP_COVERITY" ] && [ -z "$SKIP_COV_BUILD" ]; then
+        build_with_coverity
+    fi
+
+    # Run Coverity analysis
+    if [ -z "$SKIP_COVERITY" ]; then
+    # shellcheck disable=SC2086
+        if ! run_coverity_analysis $COV_ANALYZE_ARGS; then
+            echo "Coverity analysis failed"
+            return 1
+        fi
+    fi
+
+    echo "Uploading to synopsys main coverity server"
+    /auto/sw_tools/Commercial/Synopsys/Coverity/latest/linux_x86_64/bin/cov-commit-defects --ssl --on-new-cert trust --host coverity.mellanox.com --port 8443 --user "${UCC_USERNAME}" --password "${UCC_PASSWORD}" --dir "$COV_BUILD_DIR" --stream ucc_master
```

Comment:
Can we split the line into 80/120 symbols peaces for readability?

Reply:
splitted